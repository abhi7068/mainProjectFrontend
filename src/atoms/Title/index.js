import React from 'react';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';
function Title({ children }) {
  return (
    <Typography component="h1" variant="h5">
      {children}
    </Typography>
  );
}
Title.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
};
export default Title;
