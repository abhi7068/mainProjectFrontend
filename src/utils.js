import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
