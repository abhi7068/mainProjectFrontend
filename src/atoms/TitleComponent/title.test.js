import React from 'react';
import { render } from '../../utils';
import Title from './index';

test('render', () => {
  const { asFragment } = render(<Title />);
  expect(asFragment()).toMatchSnapshot();
});
test('render', () => {
  const { getByTestId } = render(<Title children="children" />);
  expect(getByTestId('Typography')).toHaveTextContent('children');
});
