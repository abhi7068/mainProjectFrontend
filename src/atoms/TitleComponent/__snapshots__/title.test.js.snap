// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<DocumentFragment>
  <h2
    class="MuiTypography-root MuiTypography-h5"
    data-testid="Typography"
    style="font-size: 29px; font-weight: 500;"
  />
</DocumentFragment>
`;
